scan.go

```diff
@@ -157,6 +165,9 @@ func (s *ScanService) Do() (*ScanCursor, error) {
 	if s.size != nil && *s.size > 0 {
 		params.Set("size", fmt.Sprintf("%d", *s.size))
 	}
+	if s.fields != nil {
```

Comment:
Maybe better to use `len(s.fields) > 0` here? If you use `service.Fields()` (no args), then it would result in `?fields=` in the query string. I don't know if ES would ignore that.